src/MailerInterface.php

```diff
@@ -60,14 +60,13 @@ public function send(MessageInterface $message): void;
      * Sends multiple messages at once.
      *
      * This method may be implemented by some mailers which support more efficient way of
-     * sending multiple messages in the same batch. It must collects failed messages and store
-     * corresponding exceptions by {@see MessageInterface::withError()}, then returns them.
+     * sending multiple messages in the same batch.
      *
      * @param MessageInterface[] $messages List of email messages, which should be sent.
      *
-     * @return MessageInterface[] List of fails messages.
+     * @return MultipleSendResult The result object that contains all messages and errors for failed sent messages.
      */
-    public function sendMultiple(array $messages): array;
+    public function sendMultiple(array $messages): MultipleSendResult;
```

Comment:
How should I know which result corresponds to which message? Array keys?

Reply:
You get the same instances of message in `$successMessages` and `$failMessages` arrays.